# Customer satisfaction analysis

![Thumbnail](https://www.corporatevision-news.com/wp-content/uploads/2022/10/Customer-Segmentation.jpg)

This project is focused on customer satisfaction for supermarkets, a dynamic and expanding sector. Examining customer demographics and purchasing behaviors, I can gain valuable insights into how different customer groups influence sales and profitability. This analysis is crucial for refining marketing strategies and enhancing customer engagement, making it especially useful for supermarkets or any product and customer based business for that matter looking to optimize their operations.

## 1. Data collection

The (CSV) dataset that I analysed in this project comes from [Aung Pyae](https://www.kaggle.com/datasets/aungpyaeap/supermarket-sales). The tools for this project are MySQL for cleaning, querying and extracting data, ran in the visual studio code integrated developement environment, and Tableau for visualizing data.

In [3]:
# First is to understand the structure of the data to create an appropriate table for it and become familiar with its content

# loads the pandas library
import pandas as pd

# reads the data
df = pd.read_csv ('https://raw.githubusercontent.com/MiguelBlair/SQL_and_Tableau_customer_satisfaction_analysis/refs/heads/main/Raw_data/supermarket_sales.csv')

# previews the data
df

Invoice ID Branch       City Customer type  Gender  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               Product line  Unit price  Quantity   Tax 5%      Total  \
0         Health and beauty       74.69         7  26.1415   548.9715   
1    Electronic accessories       15.28         5   3.8200    80.2200   
2        Home and lifestyle       46.33         7  16.2155   340.5255   
3         Health and beauty       58.22         8  23.2880   489.0480   
4         Sports and travel       86.31         7  30.2085   634.3785   
..                      ...         ...       ...      ...        ...   
995       Health and beauty       40.35         1   2.0175    42.3675   
996      Home and lifestyle       97.38        10  48.6900  1022.4900   
997      Food and beverages       31.84         1   1.5920    33.4320   
998      Home and lifestyle       65.82         1   3.2910    69.1110   
999     Fashion accessories       88.34         7  30.9190   649.2990   

          Date   Time      Payment    cogs  gross margin percentage  \
0     1/5/2019  13:08      Ewallet  522.83                 4.761905   
1     3/8/2019  10:29         Cash   76.40                 4.761905   
2     3/3/2019  13:23  Credit card  324.31                 4.761905   
3    1/27/2019  20:33      Ewallet  465.76                 4.761905   
4     2/8/2019  10:37      Ewallet  604.17                 4.761905   
..         ...    ...          ...     ...                      ...   
995  1/29/2019  13:46      Ewallet   40.35                 4.761905   
996   3/2/2019  17:16      Ewallet  973.80                 4.761905   
997   2/9/2019  13:22         Cash   31.84                 4.761905   
998  2/22/2019  15:33         Cash   65.82                 4.761905   
999  2/18/2019  13:28         Cash  618.38                 4.761905   

     gross income  Rating  
0         26.1415     9.1  
1          3.8200     9.6  
2         16.2155     7.4  
3         23.2880     8.4  
4         30.2085     5.3  
..            ...     ...  
995        2.0175     6.2  
996       48.6900     4.4  
997        1.5920     7.7  
998        3.2910     4.1  
999       30.9190     6.6  

[1000 rows x 17 columns]

In [4]:
#reads the data types
df.dtypes

Invoice ID                  object
Branch                      object
City                        object
Customer type               object
Gender                      object
Product line                object
Unit price                 float64
Quantity                     int64
Tax 5%                     float64
Total                      float64
Date                        object
Time                        object
Payment                     object
cogs                       float64
gross margin percentage    float64
gross income               float64
Rating                     float64
dtype: object

Now knowing all the columns and data types, I created a table to import the data
![img](Images/create_table.png)

Then I loaded in the data
![img](Images/load_data.png)

In SQL, to view the data types and the amount of columns, I would use the DESCRIBE query or access the table in SQL tools in VS code
![img](Images/data_types.png)

>What this preview tells us about this data-

1. There are 17 columns.

2. The columns in the datasets are: Invoice_ID, Branch, City, Customer_type, Gender, Product_line, Unit_price, Quantity, Tax_5, Total, Date, Time, Payment, cogs, gross_margin_percentage, gross_income, Rating

3. Most of the data types of the columns are appropriate for this analysis, other will need to be set.

## 2. Defining the problems/ goals

> Previewing the column and data types, the ideal key insights to extract would be-

1.What are the sales trends over time by product line? 
 - Aggregate total sales and analyze trends for different product lines over time.
 
2.What is the average purchase amount and frequency by customer type and gender? 
 - Compute average transaction value and count of transactions for each customer type and gender.

3.Which product lines or branches have the highest profit margins? 
 - Calculate profit margins (gross margin as a percentage of total) for each product line and branch to find the highest ones.
 
4.How does the payment method impact the average transaction value? 
 - Compare average transaction values across different payment methods to determine if there are significant differences.

## 3. Data cleaning

This data is fairly clean, it just needs appropriate data types set, and a "safe" copy of the original table for data integrity and to prevent data loss, also this table will only have the relevant columsn to extract the insights indicated above.

Here I created the safe copy with only the relevant columns
![img](Images/safe_copy.png)

Here I checked for duplicate rows
![img](Images/dups.png)
The result set is empty, therefore no duplicates

Here I checked for null values
![img](Images/nulls.png)
The result set is empty, therefore no null values

Here I changed the Date column to it's appropriate data type
![img](Images/date_type.png)

The data is now clean, the appropriate data types are set, those are;

Branch: VARCHAR

Rating: FLOAT

Product_line: VARCHAR

Total: FLOAT

Date: DATE

Payment: VARCHAR

gross_margin_percentage: FLOAT

gross_income: FLOAT

I created a 'safe' copy of the data with only the relevant columns, checked for duplicate and null values, and assigned appropriate data type to the Date column. Here I started extracting insights and visualizing them.

## 4. Data analysis

**First question-**

  1. What are the sales trends over time by product line?
 - Aggregate total sales and analyze trends for different product lines over time.


I ran this query to aggregate the columns
![Image](Images/q1.png)

[![Sales over time viz](https://public.tableau.com/shared/CXJDQ8TNF?:display_count=n&:origin=viz_share_link)](https://public.tableau.com/shared/CXJDQ8TNF?:display_count=n&:origin=viz_share_link)


>Conclusion

From the data on total sales across different product lines for the first three months of 2019, we can identify several key trends:

1. **Top Performing Product Lines:**
 - **Sports and Travel** consistently recorded high sales, peaking in January 2019 and maintaining strong performance throughout the quarter, showing a sustained demand in this category.
 - **Home and Lifestyle** also showed strong sales, particularly in March, where it had the highest total sales among all product lines.

2. **Product Lines with Steady Sales:**
 - **Food and Beverages** showed relatively consistent sales across the three months, though it experienced a slight drop in March compared to January and February.
 - **Fashion Accessories** remained stable but with a slight decline in March.

3. **Fluctuations in Sales:**
 - **Health and Beauty** saw notable fluctuations, with the highest sales in March but a dip in February. It suggests possible seasonal or promotional influences.
 - **Electronic Accessories** sales peaked in January and gradually declined by March, indicating reduced interest or market saturation.

Overall, the data reflects that **Sports and Travel** and **Home and Lifestyle** are the strongest performers, while **Health and Beauty** and **Electronic Accessories** experience more variability in sales.

**Second question-**

  2. Which categories have the highest-rated products?
Investigate which product categories receive the highest ratings overall. This could help identify top-performing categories.

This query accomplishes this-

![Image](Images/q2.png)

[![Average rating viz](https://public.tableau.com/shared/MD6YM46KT?:display_count=n&:origin=viz_share_link)](https://public.tableau.com/shared/MD6YM46KT?:display_count=n&:origin=viz_share_link)

>Conclusion

 The top-rated product category is **Food and Beverages**, with an average rating of 7.11, indicating a strong customer preference for these products. **Fashion Accessories** and **Health and Beauty** also perform well, with average ratings of 7.03 and 7.00, respectively. While **Electronic Accessories** (6.92) and **Sports and Travel** (6.91) are slightly lower, they still maintain competitive ratings. These insights suggest that focusing on **Food and Beverages** and **Fashion Accessories** could yield the highest customer satisfaction.

**Third question**

3. Which product lines or branches have the highest profit margins?
 - Calculate profit margins (gross margin as a percentage of total) for each product line and branch to find the highest ones.

![Image](Images/q3.png)

[![Profit margin viz](https://public.tableau.com/views/SQLandTableauCustomersatisfactionanalysis/Story1?:language=en-US&:sid=&:redirect=auth&:display_count=n&:origin=viz_share_link)](https://public.tableau.com/views/SQLandTableauCustomersatisfactionanalysis/Story1?:language=en-US&:sid=&:redirect=auth&:display_count=n&:origin=viz_share_link)


>Conclusion:

The analysis shows that all product lines and branches have a consistent profit margin percentage of approximately **4.76%**, with no significant variation across branches or product lines. 

This uniformity suggests that the company applies a standardized pricing or cost structure across different categories and locations. As a result, none of the product lines or branches stand out with particularly higher profit margins. Further analysis could focus on identifying ways to increase profitability in specific branches or product lines through cost reduction or pricing optimization.

**Fourth question-**

4. How does the payment method impact the average transaction value?
 - Compare average transaction values across different payment methods to determine if there are significant differences.

![Image](Images/q4.png)

[![Average transaction viz](https://public.tableau.com/shared/SZF4ZYJ6M?:display_count=n&:origin=viz_share_link)](https://public.tableau.com/shared/SZF4ZYJ6M?:display_count=n&:origin=viz_share_link)


>Conclusion:

From the results, we can observe that the payment method has a noticeable impact on the average transaction value:

- **Cash** has the highest average transaction value at **326.18**.
- **Credit card** follows closely with an average transaction value of **324.01**.
- **Ewallet** has the lowest average transaction value at **318.82**.

Although the differences are not drastic, cash transactions tend to have slightly higher values compared to credit cards and ewallets. This could suggest that customers paying with cash might be purchasing more expensive items or in larger quantities. Conversely, ewallet users may be making more frequent, smaller purchases. Further analysis could explore additional factors such as customer demographics or product categories to explain these differences.

# Summary:

The analysis of the dataset provides insights into key factors impacting sales, customer satisfaction, profitability, and payment methods.

1. **Sales Trends Over Time by Product Line:**
 - **Conclusion:** 
 - **Sports and Travel** and **Home and Lifestyle** are the top-performing product lines, with **Sports and Travel** peaking in January and **Home and Lifestyle** showing the strongest sales in March.
 - **Health and Beauty** and **Electronic Accessories** exhibited fluctuations, with **Health and Beauty** seeing a notable increase in March and **Electronic Accessories** experiencing a decline over time.
 - **Food and Beverages** and **Fashion Accessories** remained stable, with minor fluctuations in sales.

2. **Highest-Rated Product Categories:**
 - **Conclusion:** 
 - The top-rated product categories are:
 - **Food and Beverages** – Average Rating: 7.11
 - **Fashion Accessories** – Average Rating: 7.03
 - **Health and Beauty** – Average Rating: 7.00
 - These categories receive the highest ratings, indicating strong customer satisfaction. Although **Electronic Accessories** (6.92) and **Sports and Travel** (6.91) are slightly lower, they still maintain competitive ratings.

3. **Profit Margins by Product Line and Branch:**
 - **Conclusion:** 
 - The profit margins for all product lines and branches are consistent at approximately **4.76%**, suggesting a standardized pricing or cost structure across categories and locations. No product line or branch significantly stands out in terms of profitability.

4. **Impact of Payment Method on Average Transaction Value:**
 - **Conclusion:** 
 - **Cash** transactions have the highest average transaction value at **₹326.18**, followed by **Credit Card** at **₹324.01**, and **Ewallet** at **₹318.82**. While the differences are subtle, cash transactions tend to involve slightly larger or more expensive purchases, whereas ewallet users may make smaller, more frequent purchases.

Overall, the analysis highlights strong sales trends in certain product lines, high customer satisfaction in specific categories, consistent profit margins across branches and product lines, and noticeable differences in transaction values based on payment methods.

### Thank you for your time

If you have any ideas or suggestions for this project, feel free to make those changes or just comment it.